File: ltt_client/src/store/models/article.ts
import {UserBasicInfo} from "@/store/models/user";

export class Article{
    constructor(
        public id:number,
        public title:string,
        public content:string,
        public author_id:number,
        public create_time:string,
        public edit_time:string,
        public tag_ids:number[]
    ) {
    }
    static emptu(){
        return new Article(
            -1,"","",-1,"",
            "",[]
        )
    }
}

export class ArticlePreview{
    constructor(
        public id:number,
        public title:string,
        public preview:string,
        public author_id:number,
        public create_time:string,
        public edit_time:string,
        public tag_ids:number[]
    ) {
    }
}
// export class PageOfArticle{
//     constructor(
//         public pageindex:number,
//         public articles:ArticlePreview[]
//     ) {
//     }
//     static testpre(){
//         let arr=[]
//         for(let i=0;i<10;i++){
//             arr.push(
//                 new ArticlePreview(
//                     0,
//                     "title",
//                     "content",
//                     new UserBasicInfo(
//                         "hhh",
//                         "",""
//                     )
//                 )
//             )
//         }
//         return new PageOfArticle(
//             0,
//             arr
//         )
//     }
// }

export class ArticlePreviewMap{
    private _map:any={}
    insert(article:ArticlePreview){
        this._map["$"+article.id]=article
    }
    getbyid(id:number):ArticlePreview{
        return this._map["$"+id]
    }
    constructor() {
    }
}

File: ltt_server/src/models/article.rs
use crate::models::user::{UserId, UserSimpleInfo};
use serde::{Deserialize, Serialize};

pub type ArticleId=i64;

// article_info table in sql
/*
文章存储在本地的命名格式: 日期_文章id
 */

#[derive(Default,Deserialize, Serialize)]
pub struct Article{
    pub id:ArticleId, // 文呼唤你u章id,用于在其他表里检索信息
    pub title:String,  // 文章标题
    pub content:String,
    // pub file_path:String,  // 文章存储在本地的位置
    pub author_id:UserId,  // 作者id，user表中的id
    pub create_time : String,  // 发布日期
    pub edit_time:String,
    pub tag_ids : Vec<i32> // 标签，至多有三个
}

//列表内 文章预览内容
#[derive(Default,Deserialize, Serialize)]
pub struct ArticlePreview{
    pub id:ArticleId, // 文呼唤你u章id,用于在其他表里检索信息
    pub title:String,  // 文章标题
    pub preview:String,
    // pub file_path:String,  // 文章存储在本地的位置
    pub author_id:UserId,  // 作者id，user表中的id
    pub create_time : String,  // 发布日期
    pub edit_time:String,
    pub tag_ids : Vec<i32> // 标签，至多有三个
}

//一页文章预览
pub struct PageOfArticle{
    pub pageindex:u32,
    pub articles:Vec<ArticlePreview>
}